### ISSUE TITLE
8. Due to the way vectors are stored internally, both "1" and 1 are treated as the same index. This is a bug.

### ISSUE THREAD

**Issue by Lartu** [dvkt/dino#6]
This is incredible. You just pulled a complete LDPL interpreter, written in LDPL that has its own bytecode out of your sleeve in the blink of an eye. I don't really know what's more impressive of that. My infinite respects.

> 8. Due to the way vectors are stored internally, both "1" and 1 are
   treated as the same index. This is a bug.

In the readme you say this. Is that really a bug? In LDPL, if I recall correctly (I can't find this in the documentation so I'll get to it asap) `myVector:1` is silently turned into `myVector:"1"`. If that's the case, this bug is not a bug but a feature!

**Comment by Lartu** [comment]
Yes, look at how ldpl_vectors are implemented:

```c++
template<typename T>
struct ldpl_vector {
    unordered_map<string, T> inner_map;
    T& operator [] (const string& i) {
        return inner_map[i];
    }
    T& operator [] (ldpl_number i) {
        return inner_map[to_ldpl_string(i)];
    }
    void clear(){
        inner_map.clear();
    }
    ldpl_number count(){
        return inner_map.size();
    }
};
```

Number subindices are turned into strings!

**Comment by xvxx** [comment]
>  If that's the case, this bug is not a bug but a feature! [...] Number subindices are turned into strings!

Oh, amazing! Thank you for checking.  Fixed in d84e788f62ed999d8ae3d7818e6a912d31edaade

> My infinite respects.

Thank you! I had been wanting to do something like this since I first saw the language. It's impractically slow, but maybe it shows that an LDPL compiler could be built using LDPL someday! Dino's lexer and parser could be used as a trial run of something like that.

**Comment by Lartu** [comment]
> Oh, amazing! Thank you for checking. Fixed in d84e788

:wink: 

> Thank you! I had been wanting to do something like this since I first saw the language. It's impractically slow, but maybe it shows that an LDPL compiler could be built using LDPL someday! Dino's lexer and parser could be used as a trial run of something like that.

Awesome, haha! Something wonderful could come from dino, I'll be keeping an eye on what you do with it :smile:
